src/OpenSage.Game/Data/W3d/W3dCollectionHeader.cs

```diff
@@ -33,7 +35,7 @@ internal static W3dCollectionHeader Parse(BinaryReader reader, W3dParseContext c
         protected override void WriteToOverride(BinaryWriter writer)
         {
             writer.Write(Version);
-            writer.Write(Name);
+            writer.WriteFixedLengthString(Name, NameSize);
```

Comment:
Sorry - I should have been clearer. No need to store `NameSize`. You can do

``` csharp
writer.WriteFixedLengthString(Name, Name.Length + 1);
```

This way of doing it has come up a few times in w3d chunks now, so we should probably have a helper method for it.